File: others/client/Client 1.0/App/Core/Message.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Genting.Infrastructure.CommonServices.Client.Core
{
    public class Message : IMessage
    {
        public string InterpreterType { get; set; }

        public string MessageType { get; set; }

        public string Payload { get; set; }
       
        public static IMessage CreateMessage(string interpreterType, object info, string messegeType = "default")
        {
            Message msg = new Message() { InterpreterType = interpreterType, MessageType = messegeType };
            msg.Payload = JsonSerializer.Instance.Serialize(info);
            return msg;
        }
    }
}

File: others/client/Client 1.0/Angular/core/Message.ts
import { IMessage } from "./IMessage";
import { retry } from "rxjs/operators/retry";

export class Message implements IMessage {
    InterpreterType: string;
    MessageType: string;
    Payload: string;

    public static Parse<T>(payload: string): T {
        return <T>JSON.parse(payload);
    }

    public ToString(): string {
        return JSON.stringify(this);
    }
}